retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/ArmeriaCallFactory.java

```diff
@@ -38,24 +43,37 @@
 /**
  * A {@link Call.Factory} that creates a {@link Call} instance for {@link HttpClient}.
  */
-public class ArmeriaCallFactory implements Call.Factory {
+final class ArmeriaCallFactory implements Call.Factory {
 
-    private final HttpClient httpClient;
+    private final Map<String, HttpClient> httpClients = new ConcurrentHashMap<>();
```

Comment:
But `@Url` annotation may connect to different authority.
https://square.github.io/retrofit/2.x/retrofit/retrofit2/http/Url.html